### PR TITLE
[INLONG-8590][Manager] Make Cluster and ClusterTags as public resources

### LINKED ISSUE #8590: [Improve][Manager] Make Cluster and ClusterTags as public resources
### Description

Now we treat Cluster and ClusterTags as the tenant resources.  Which leads to a very complicated resource-sharing policy. 

In order to achieve cluster sharing, we have to register each cluster/cluster tag into each tenant. This leads to a modification storm when any cluster changes its metadata, such as access point or IP port.

The improved method is to make cluster and cluster tags as public resources.  A new table to register the cluster tags used by each tenant is necessary.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8590 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantClusterTagEntity.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.dao.entity;
+
+import java.io.Serializable;
+import java.util.Date;
+
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantClusterTagEntity.java:22]: `@Data`

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantClusterTagEntity.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+public class TenantClusterTagEntity implements Serializable {
```
Comment (fuweng11) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantClusterTagEntity.java:23]: Maybe `RelationTenantClusterEntity`.

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantClusterTagEntity.java (hunk 1 above)
Comment (vernedeng) [reply inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantClusterTagEntity.java:22]: fixed, thx

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantClusterTagEntity.java (hunk 2 above)
Comment (vernedeng) [reply inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/TenantClusterTagEntity.java:23]: No need to add additional words to indicate that it's a relation table to tenant and cluster.

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-web/sql/changes-1.9.0.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+-- When upgrading to version 1.9.0, please execute those SQLs in the DB (such as MySQL) used by the Manager module.
+
+SET NAMES utf8mb4;
+SET FOREIGN_KEY_CHECKS = 0;
+
+USE `apache_inlong_manager`;
+
+-- To improve the ability of resources sharing. More details please see #8590
+ALTER TABLE `inlong_cluster` DROP COLUMN `tenant`;
+DROP INDEX cluster_tenant_index ON inlong_cluster;
```
Comment (healchow) [inline inlong-manager/manager-web/sql/changes-1.9.0.sql:28]: Does the `drop index` operation need to be executed before the `drop column`?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/InlongRoleService.java [hunk 4]
```diff
@@ -34,5 +34,4 @@ public interface InlongRoleService {
     InlongRoleInfo get(int id);
 
     InlongRoleInfo getByUsername(String username);
-
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/InlongRoleService.java:37]: Unnecessary change.

File: inlong-manager/manager-web/sql/changes-1.9.0.sql (hunk 3 above)
Comment (vernedeng) [reply inlong-manager/manager-web/sql/changes-1.9.0.sql:28]: You are right, I will fix it, thx.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/InlongRoleService.java (hunk 4 above)
Comment (vernedeng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/user/InlongRoleService.java]: fixed, thx

**Review by dockerzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]
